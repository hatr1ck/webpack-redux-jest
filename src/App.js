import React, { Component } from "react";
import ReactDOM from "react-dom";
import { connect } from 'react-redux';
import { mockAPI, add } from './actions';
import './App.css';

export class App extends Component {
    state = {
        username: 'dd',
        testState:'',
        test:""
    }

    write = (e) => {
        e.preventDefault()
        this.setState({
            username: e.target.value
        })
    }

    click = (e) => {
        e.preventDefault()
        this.props.mockAPI(this.state.username)
        this.setState({
            username: ''
        })
    }
    componentDidMount(){
      this.setState({
        test:'LOOL'
      })
    }
    render() {
        return (
            <div id="article-div">
      <button  onClick={()=>this.props.add('KEKISTAN)))')}>ADD</button>
       <form onSubmit={this.click}>
      <input  value={this.state.username} onChange={this.write.bind(this)} />
       <div id='kurwa'>{this.state.test}</div>
      <button id='clicks' onClick={this.click}>ADD FROM API</button>
    </form>
      <ul>
        {this.props.reducer2.map((elem, id)=>{
          return (<h1 key={id}><img src={elem.ava}/>{elem.username}</h1>)
        })}
      </ul>
      <div id="checkState">{this.state.testState}</div>
      <div id="props">{this.props.reducer1}</div>
      </div>
        );
    }
}

export default connect(state => { return state }, {
    mockAPI,
    add
})(App);