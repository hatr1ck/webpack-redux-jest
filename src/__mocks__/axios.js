let get = function(url){
	return Promise.resolve({
      status: 'success',
      data: {data:{avatar:'TestAva'}},
    })
}
exports.get = get;