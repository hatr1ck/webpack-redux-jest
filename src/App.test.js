import React from 'react';
import {App} from './App';
import toJson from 'enzyme-to-json';
import axios from 'axios'
import Adapter from 'enzyme-adapter-react-16';
import Enzyme, {shallow, render, mount} from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

  	test('taking snapshot?', ()=>{
  		let component = render(<App reducer2={[1,2,3]} />)
  	expect(toJson(component)).toMatchSnapshot()
  	})

    test('foofoo', () => {
  	let component = shallow(<App reducer2={[1,2,3]}  reducer1={'Initial State'} />)
  	expect(component.find('#props').text()).toEqual('Initial State')
    });

    test('testing input change', () => {
    let wrapper = shallow(<App reducer2={[1,2,3,4]}/>);
    let instance = wrapper.instance();
    let mockPrevent = jest.fn()
    instance.write({target:{value:'Kevin'},preventDefault:mockPrevent})
    expect(wrapper.state('username')).toBe('Kevin');
  });
    
    test('testing PreventDefault', () => {
    let wrapper = shallow(<App reducer2={[1,2,3,4]}/>);
    let instance = wrapper.instance();
    let mockPrevent = jest.fn()
    instance.write({target:{value:'Kevin'},preventDefault:mockPrevent})
    expect(mockPrevent).toHaveBeenCalled();


  });

    test('check how state is rendered', ()=>{
    let wrapper = shallow(<App reducer2={[1,2,3,4]}/>);
    wrapper.setState({testState:"str"})
    expect(wrapper.find('#checkState').text()).toBe("str")
    })
    
    test('check how state is rendered', ()=>{
    let wrapper = shallow(<App reducer2={[1,2,3,4]}/>);
    wrapper.setState({testState:"str"})
    expect(wrapper.find('#checkState').text()).toBe("str")
    })
