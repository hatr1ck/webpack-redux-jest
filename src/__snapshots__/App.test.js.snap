// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`taking snapshot? 1`] = `
<div
  id="article-div"
>
  <button>
    ADD
  </button>
  <form>
    <input
      value="dd"
    />
    <div
      id="kurwa"
    />
    <button
      id="clicks"
    >
      ADD FROM API
    </button>
  </form>
  <ul>
    <h1>
      <img />
    </h1>
    <h1>
      <img />
    </h1>
    <h1>
      <img />
    </h1>
  </ul>
  <div
    id="checkState"
  />
  <div
    id="props"
  />
</div>
`;
